### ISSUE TITLE
Reload while sprinting/Parkour Aced and Lock and Load animation bugs

### ISSUE THREAD

**Issue by Siellenexia** [HugoZink/IreNFist#40]
[The first bug](https://youtu.be/oz_dnl0pv9Q) can occur whenever I cancel the reloading while sprinting and is quite common. The sprinting animation stops if I sprint or shoot again, as you can see in the video. This doesn't happen if I don't have Parkour skill Aced.

The second bug triggers the Lock and Load animation/Hip fire sprinting animation, even though I don't have the skill. And it is quite rare and was difficult to reproduce. Both bugs happen with every weapon available in the game.

I tried removing some mods that might be the perpetrator of those bugs, but the problem still persists. 
It would be great if you can look further into this. Thank you.

**Comment by HugoZink** [comment]
Thanks for the detailed report with video, it's very much appreciated! I'll get on fixing this.

**Comment by Siellenexia** [comment]
Thank you for your response and for addressing the issue. o7

**Comment by HugoZink** [comment]
[This issue is potentially fixed on the experimental branch, check it out!](https://github.com/HugoZink/IreNFist/archive/experimental.zip) Extract the "IRE AND FIST REBORN" folder to your PAYDAY 2/mods folder. It might be best to remove the existing folder first.

**Comment by HugoZink** [comment]
Interesting, looks like this can only happen as soon as either half or all of the reload timer has elapsed (and the animation hasn't finished playing yet). Happens as soon as bullets into your mag get refilled. This might be a tricky one.

**Comment by HugoZink** [comment]
This should now be fixed.

**Comment by HugoZink** [comment]
I still get the issue sometimes but not in a way I can easily reproduce. If you can find a specific way it would be appreciated.

**Comment by Siellenexia** [comment]
Two possibilities here:
1. On the first video and at 33 seconds mark, instead of playing the reload animation while sprinting, it played the normal sprinting animation instead. And the animation gets stuck until I either shoot or sprint again. This occurs rarely, and it was difficult for me to reproduce as well.
2. It's difficult for me to explain, but I've recorded several clips with (at least) coherent explanations inside. I will upload them soon.

**Comment by Siellenexia** [comment]
The video is currently on upload and the bugs may be tricky, but I've managed to reproduce them. I already explained the bug in those clips, but I'll say here anyway:
In vanilla Payday 2, if you have Parkour Aced skill, there are three animation "switches" or phases:
1. Reload animation
2. Static animation
3. Sprint animation
The bug occurs if you release the sprint key on the second animation, or a few seconds after the reload animation. I've included a freeze frame and a slow motion in the video for this one. 
However, if you keep holding the sprint button a little longer after either animation, this bug does not occur.

**Comment by Siellenexia** [comment]
Here's the video:
https://www.youtube.com/watch?v=ETP5MFkx9K4
I apologize for the poor editing and/or quality, but I hope it's adequate to narrow it down.

**Comment by HugoZink** [comment]
This is proving to be a bit harder to fix than I thought. I'm still working on it, but I may have to drop the priority just a tad since it's harder to trigger now and isn't gamebreaking.

**Comment by Siellenexia** [comment]
Yeah, it's perfectly fine and fair--not to mention the sprinting bug only occurs if you have Parkour skill Aced.

**Comment by Siellenexia** [comment]
I've found a way to reproduce the Lock and Load animation bug, but you have to be (somehow) precise; The bug occurs if you press and hold the sprint key as soon as the reload finishes.
Here's a video of the example:
https://youtu.be/qAVUzi6iKXg

**Comment by HugoZink** [comment]
I've done quite a bit of testing in the code so far, but no luck actually fixing it. InF doesn't/shouldn't touch anything related to the stances anymore, other than allowing running while holding melee.

**Comment by Siellenexia** [comment]
It seems that running while holding melee can trigger the Lock and Load animation bug.

**Comment by artemisexe** [comment]
A new note:
This doesn't happen on the weapons from the Gunslinger Pack.
